include/wasm/WasmModule.h

```diff
@@ -86,12 +90,20 @@ class WasmModule
     void clearCapturedStdout();
 
     // ----- Memory management -----
-    virtual uint32_t mmapMemory(uint32_t length);
+    uint32_t getCurrentBrk();
+
+    virtual uint32_t growMemory(uint32_t nBytes);
+
+    virtual void shrinkMemory(uint32_t nBytes);
 
-    virtual uint32_t mmapPages(uint32_t pages);
+    virtual uint32_t mmapMemory(uint32_t nBytes);
 
     virtual uint32_t mmapFile(uint32_t fp, uint32_t length);
 
+    virtual void unmapMemory(uint32_t offset, uint32_t nBytes);
+
+    uint32_t createMemoryGuardRegion();
+
```

Comment:
Hopefully the `WasmModule` interface can become more and more generic so we can share more functionality between WAMR and WAVM (and VMs we add in future...)